src/set_theory/zfc.lean

```diff
@@ -371,7 +384,7 @@ open pSet
 /-- Turns a pre-set into a ZFC set. -/
 def mk : pSet → Set := quotient.mk
 
-@[simp] theorem mk_eq (x : pSet) : @eq Set ⟦x⟧ (mk x) := rfl
+@[simp] theorem mk_eq (x : pSet) : ⟦x⟧ = mk x := rfl
```

Comment:
Even with `pp.notation false`?

Reply:
Yes.

Reply:
Doesn't look like it:
```lean
@[simp] theorem mk_eq (x : pSet) : ⟦x⟧ = mk x := rfl
set_option pp.all true
#print mk_eq
```
```lean
@[_refl_lemma, _simp_cache, simp]
theorem Set.mk_eq : ∀ (x : pSet.{u_1}),
  @eq.{u_1+2} (@quotient.{u_1+2} pSet.{u_1} pSet.setoid.{u_1}) (@quotient.mk.{u_1+2} pSet.{u_1} pSet.setoid.{u_1} x)
    (Set.mk.{u_1} x) :=
λ (x : pSet.{u_1}),
  @rfl.{u_1+2} (@quotient.{u_1+2} pSet.{u_1} pSet.setoid.{u_1}) (@quotient.mk.{u_1+2} pSet.{u_1} pSet.setoid.{u_1} x)
```

Reply:
Ah, I didn't enable `pp.all`. That's bewildering. Is this discrepancy something we care about, though?

Reply:
This PR is conflicting with #15324 which also made this equality change. It should be reverted.

Reply:
> Is this discrepancy something we care about, though?

Based on the statement of the theorem, clearly this was the intent. The type of the equality can sometimes affect whether rewrites succeed, and this one infers a weird type because the quotient.mk application on the left is not "type correct" in the sense of using the redefined notation (this is the entire point of the redefinition - `quotient.mk` messes tactics up because it has the wrong type).

Reply:
I'll revert it, but I still want to know: why do these not generate the same terms, and why do we care?